File: backend/src/main/scala/microservices/traveler/objects/TravelerListPlannerResponse.scala
// 本文件定义 traveler 模块的列表响应对象 `TravelerListPlannerResponse`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TravelerListPlannerResponse(
    travelers: List[TravelerPlannerResponse]
)

object TravelerListPlannerResponse:
  given sourceEncoder: Encoder[TravelerListPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[TravelerListPlannerResponse] = deriveDecoder

File: frontend/src/microservices/traveler/objects/TravelerListPlannerResponse.ts
// 本文件定义 traveler 模块的 `TravelerListPlannerResponse`，作为列表响应数据并提供 JSON 编解码。

import type { TravelerPlannerResponse } from './TravelerPlannerResponse'

export type TravelerListPlannerResponse = {
  travelers: TravelerPlannerResponse[]
}

export const travelerListPlannerResponseFromJson = (json: string): TravelerListPlannerResponse =>
  JSON.parse(json) as TravelerListPlannerResponse

export const travelerListPlannerResponseToJson = (value: TravelerListPlannerResponse): string =>
  JSON.stringify(value)
